.github/workflows/build-and-release.yaml

```diff
@@ -0,0 +1,103 @@
+name: Docker build and Release
```

Comment:
This doesn't seem like `build and release` step since it is only catering PRs.

Reply:
It creates a release also on a successful push on master

Reply:
Changed in the separation of workflow files